dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -306,6 +306,11 @@ impl ResourceDiscovery for CommandDiscovery {
                     }
                 }
                 self.adapters = adapters;
+                // go through the resource and remove duplicates with same name and version
```

Comment:
Is the intent of `insert_resource()` to take care of duplicates when `skip_duplicate_version` is true (as it is on L.303)?

I think https://github.com/PowerShell/DSC/blob/main/dsc_lib/src/discovery/command_discovery.rs#L570 
should be `if skip_duplicate_version` instead of  `if !skip_duplicate_version`

Reply:
Good catch, let me check on this